cpg-language-configfiles/src/main/kotlin/de/fraunhofer/aisec/cpg/frontend/configfiles/IniFilesFrontend.kt

```diff
@@ -0,0 +1,158 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ *                    $$$$$$\  $$$$$$$\   $$$$$$\
+ *                   $$  __$$\ $$  __$$\ $$  __$$\
+ *                   $$ /  \__|$$ |  $$ |$$ /  \__|
+ *                   $$ |      $$$$$$$  |$$ |$$$$\
+ *                   $$ |      $$  ____/ $$ |\_$$ |
+ *                   $$ |  $$\ $$ |      $$ |  $$ |
+ *                   \$$$$$   |$$ |      \$$$$$   |
+ *                    \______/ \__|       \______/
+ *
+ */
+package de.fraunhofer.aisec.cpg.frontend.configfiles
+
+import de.fraunhofer.aisec.cpg.TranslationContext
+import de.fraunhofer.aisec.cpg.frontends.Language
+import de.fraunhofer.aisec.cpg.frontends.LanguageFrontend
+import de.fraunhofer.aisec.cpg.frontends.TranslationException
+import de.fraunhofer.aisec.cpg.graph.*
+import de.fraunhofer.aisec.cpg.graph.declarations.RecordDeclaration
+import de.fraunhofer.aisec.cpg.graph.declarations.TranslationUnitDeclaration
+import de.fraunhofer.aisec.cpg.graph.types.Type
+import de.fraunhofer.aisec.cpg.sarif.PhysicalLocation
+import de.fraunhofer.aisec.cpg.sarif.Region
+import java.io.File
+import java.io.FileInputStream
+import java.net.URI
+import org.ini4j.Ini
+import org.ini4j.Profile
+
+/**
+ * The INI file frontend. This frontend utilizes the [ini4j library](https://ini4j.sourceforge.net/)
+ * to parse the config file. The result consists of
+ * - a [TranslationUnitDeclaration] wrapping the entire result
+ * - a [de.fraunhofer.aisec.cpg.graph.declarations.NamespaceDeclaration] wrapping the INI file and
+ *   thus preventing collisions with other symbols which might have the same name
+ * - a [RecordDeclaration] per `Section` (a section refers to a block of INI values marked with a
+ *   line `[SectionName]`)
+ * - a [de.fraunhofer.aisec.cpg.graph.declarations.FieldDeclaration] per entry in a section. The
+ *   [de.fraunhofer.aisec.cpg.graph.declarations.FieldDeclaration.name] matches the `entry`s `name`
+ *   field and the [de.fraunhofer.aisec.cpg.graph.declarations.FieldDeclaration.initializer] is set
+ *   to a [statements.expressions.Literal] with the corresponding `entry`s `value`.
+ *
+ * Note:
+ * - the "ini4j" library does not provide any super type for all nodes. Thus, the frontend accepts
+ *   `Any`
+ * - [typeOf] has to be implemented, but as there are no types always returns the builtin `string`
+ *   type
+ * - [codeOf] has to accept `Any` (because of the limitations stated above) and simply returns
+ *   `.toString()`
+ * - [locationOf] always returns `null` as the "ini4j" library does not provide any means of getting
+ *   a location given a node
+ * - [setComment] not implemented as this is not used (no
+ *   [de.fraunhofer.aisec.cpg.frontends.Handler] pattern implemented)
+ * - Comments in general are not supported.
+ */
+class IniFilesFrontend(language: Language<IniFilesFrontend>, ctx: TranslationContext) :
```

Comment:
Should probably be `IniFileFrontend`, we always had a singular in the frontend, also similarly in the language

Reply:
Done